File: 3208/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(numberOfAlternatingGroups([]int{1,1,0,1}, 4))
}

func numberOfAlternatingGroups(colors []int, k int) int {
	n := len(colors)
	res, cnt := 0, 1
	for i := -k + 2; i < n; i++ {
		if colors[(i+n) % n] != colors[(i-1+n)%n] {
			cnt++
		} else {
			cnt = 1
		}
		if cnt >= k {
			res++
		}
	}
	return res
}

File: 3208/main.rs
struct Solution {}

impl Solution {
    pub fn number_of_alternating_groups(colors: Vec<i32>, k: i32) -> i32 {
        let n = colors.len() as i32;
        let mut res = 0;
        let mut cnt = 1;
        for i in (-k + 2)..n {
            if colors[((i+n) % n) as usize] != colors[((i-1+n)%n) as usize] {
                cnt+=1;
            } else {
                cnt = 1;
            }
            if cnt >= k {
                res += 1;
            }
        }
        return res
    }
}

fn main() {
    println!("{:#?}", Solution::number_of_alternating_groups(vec![1,1,0,1], 4));
}